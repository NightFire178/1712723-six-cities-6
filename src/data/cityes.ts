const Cityes = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`,
];
export default Cityes
