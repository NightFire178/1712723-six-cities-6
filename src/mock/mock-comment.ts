import mockUser from "./mock-user";
import comment from '../types/comment'

const mockComment:comment = {
  comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
  date: `2021-03-07T08:04:28.647Z`,
  id: 10,
  rating: 4,
  user: mockUser
}
export default mockComment

export const mockArrayComments:Array<comment> =[
  {
    comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 15,
    rating: 3,
    user: mockUser
  },
  {
    comment: `Beautiful space, fantastic location and atmosphere,
  really a wonderful place to spend a few days. Will be back.`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 13,
    rating: 5,
    user: mockUser
  },
  {
    comment: `What an amazing view! The house is stunning and in
    an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2021-03-07T08:04:28.647Z`,
    id: 10,
    rating: 4,
    user: mockUser
  }
]
