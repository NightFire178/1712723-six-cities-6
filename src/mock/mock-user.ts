import User from "../types/user";

const mockUser:User = {
  id: 1,
  name: `vasa`,
  is_pro: true,
  avatar_url:`https://assets.htmlacademy.ruâ€¦tar/6.jp`,
  email: `sdfhskd@maid.ru`
}
export default mockUser
