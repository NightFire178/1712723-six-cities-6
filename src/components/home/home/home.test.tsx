import React from 'react'
import {render} from '@testing-library/react'
import HomeComponents from "./home";
import cityes from "../../../data/cityes";
import {mockArrayHotels} from "../../../mock/mock-hotels";

//eslint-disable-next-line
jest.mock("../../block/header/header", ()=>()=><div>header</div>)

//eslint-disable-next-line
jest.mock("../../block/nav/nav", ()=>()=><div>nav</div>)

//eslint-disable-next-line
jest.mock("../hotels/hotels", ()=>()=><div>hotels</div>)

const appStateSelection = {
  cityNow:() =>(cityes[0])
}

jest.mock("../../../redux/selectors/app-state", ()=>appStateSelection)

const hotelSelection = {
  hotels: ()=>JSON.parse(JSON.stringify(mockArrayHotels))
}

jest.mock("../../../redux/selectors/hotels", ()=>hotelSelection)

test(`home home component test`, () => {
  const header = render(<HomeComponents/>)
  expect(header).toMatchSnapshot()
})
