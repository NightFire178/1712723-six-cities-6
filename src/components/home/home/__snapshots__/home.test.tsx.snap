// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home home component test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          style="display: none;"
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
          >
            <symbol
              id="icon-arrow-select"
              viewBox="0 0 7 4"
            >
              <path
                clip-rule="evenodd"
                d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
                fill-rule="evenodd"
              />
            </symbol>
            <symbol
              id="icon-bookmark"
              viewBox="0 0 17 18"
            >
              <path
                d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
              />
            </symbol>
            <symbol
              id="icon-star"
              viewBox="0 0 13 12"
            >
              <path
                clip-rule="evenodd"
                d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
                fill-rule="evenodd"
              />
            </symbol>
          </svg>
        </div>
        <div
          class="page page--gray page--main false"
        >
          <div>
            header
          </div>
          <main
            class="page__main page__main--index"
          >
            <h1
              class="visually-hidden"
            >
              Cities
            </h1>
            <div>
              nav
            </div>
            <div
              class="cities"
            >
              <div
                class="cities__places-container container"
              >
                <div>
                  hotels
                </div>
                <div
                  class="cities__right-section"
                >
                  <section
                    class="cities__map map"
                  >
                    <div
                      class="leaflet-container leaflet-grab leaflet-touch-drag"
                      id="map"
                      style="height: 100%; position: relative;"
                      tabindex="0"
                    >
                      <div
                        class="leaflet-pane leaflet-map-pane"
                        style="left: 0px; top: 0px;"
                      >
                        <div
                          class="leaflet-pane leaflet-tile-pane"
                        >
                          <div
                            class="leaflet-layer "
                            style="z-index: 1;"
                          >
                            <div
                              class="leaflet-tile-container leaflet-zoom-animated"
                              style="z-index: 18; left: 0px; top: 0px;"
                            >
                              <img
                                alt=""
                                class="leaflet-tile"
                                role="presentation"
                                src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
                                style="width: 256px; height: 256px; left: -130px; top: -72px;"
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="leaflet-pane leaflet-shadow-pane"
                        />
                        <div
                          class="leaflet-pane leaflet-overlay-pane"
                        />
                        <div
                          class="leaflet-pane leaflet-marker-pane"
                        >
                          <img
                            alt=""
                            class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                            src="img/pin.svg"
                            style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -64px; top: -45px; z-index: -45;"
                            tabindex="0"
                          />
                        </div>
                        <div
                          class="leaflet-pane leaflet-tooltip-pane"
                        />
                        <div
                          class="leaflet-pane leaflet-popup-pane"
                        />
                      </div>
                      <div
                        class="leaflet-control-container"
                      >
                        <div
                          class="leaflet-top leaflet-left"
                        />
                        <div
                          class="leaflet-top leaflet-right"
                        />
                        <div
                          class="leaflet-bottom leaflet-left"
                        />
                        <div
                          class="leaflet-bottom leaflet-right"
                        >
                          <div
                            class="leaflet-control-attribution leaflet-control"
                          >
                            <a
                              href="https://leafletjs.com"
                              title="A JS library for interactive maps"
                            >
                              Leaflet
                            </a>
                             | © 
                            <a
                              href="https://www.openstreetmap.org/copyright"
                            >
                              OpenStreetMap
                            </a>
                             contributors © 
                            <a
                              href="https://carto.com/attributions"
                            >
                              CARTO
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                  </section>
                </div>
              </div>
            </div>
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        style="display: none;"
      >
        <svg
          xmlns="http://www.w3.org/2000/svg"
        >
          <symbol
            id="icon-arrow-select"
            viewBox="0 0 7 4"
          >
            <path
              clip-rule="evenodd"
              d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
              fill-rule="evenodd"
            />
          </symbol>
          <symbol
            id="icon-bookmark"
            viewBox="0 0 17 18"
          >
            <path
              d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
            />
          </symbol>
          <symbol
            id="icon-star"
            viewBox="0 0 13 12"
          >
            <path
              clip-rule="evenodd"
              d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
              fill-rule="evenodd"
            />
          </symbol>
        </svg>
      </div>
      <div
        class="page page--gray page--main false"
      >
        <div>
          header
        </div>
        <main
          class="page__main page__main--index"
        >
          <h1
            class="visually-hidden"
          >
            Cities
          </h1>
          <div>
            nav
          </div>
          <div
            class="cities"
          >
            <div
              class="cities__places-container container"
            >
              <div>
                hotels
              </div>
              <div
                class="cities__right-section"
              >
                <section
                  class="cities__map map"
                >
                  <div
                    class="leaflet-container leaflet-grab leaflet-touch-drag"
                    id="map"
                    style="height: 100%; position: relative;"
                    tabindex="0"
                  >
                    <div
                      class="leaflet-pane leaflet-map-pane"
                      style="left: 0px; top: 0px;"
                    >
                      <div
                        class="leaflet-pane leaflet-tile-pane"
                      >
                        <div
                          class="leaflet-layer "
                          style="z-index: 1;"
                        >
                          <div
                            class="leaflet-tile-container leaflet-zoom-animated"
                            style="z-index: 18; left: 0px; top: 0px;"
                          >
                            <img
                              alt=""
                              class="leaflet-tile"
                              role="presentation"
                              src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
                              style="width: 256px; height: 256px; left: -130px; top: -72px;"
                            />
                          </div>
                        </div>
                      </div>
                      <div
                        class="leaflet-pane leaflet-shadow-pane"
                      />
                      <div
                        class="leaflet-pane leaflet-overlay-pane"
                      />
                      <div
                        class="leaflet-pane leaflet-marker-pane"
                      >
                        <img
                          alt=""
                          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                          src="img/pin.svg"
                          style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -64px; top: -45px; z-index: -45;"
                          tabindex="0"
                        />
                      </div>
                      <div
                        class="leaflet-pane leaflet-tooltip-pane"
                      />
                      <div
                        class="leaflet-pane leaflet-popup-pane"
                      />
                    </div>
                    <div
                      class="leaflet-control-container"
                    >
                      <div
                        class="leaflet-top leaflet-left"
                      />
                      <div
                        class="leaflet-top leaflet-right"
                      />
                      <div
                        class="leaflet-bottom leaflet-left"
                      />
                      <div
                        class="leaflet-bottom leaflet-right"
                      >
                        <div
                          class="leaflet-control-attribution leaflet-control"
                        >
                          <a
                            href="https://leafletjs.com"
                            title="A JS library for interactive maps"
                          >
                            Leaflet
                          </a>
                           | © 
                          <a
                            href="https://www.openstreetmap.org/copyright"
                          >
                            OpenStreetMap
                          </a>
                           contributors © 
                          <a
                            href="https://carto.com/attributions"
                          >
                            CARTO
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </section>
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
